py-polars/tests/unit/test_async.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+
+import asyncio
+import time
+from functools import partial
+from typing import Any, Callable
+
+import pytest
+
+import polars as pl
+from polars.dependencies import gevent
+
+
+async def _aio_collect_async(raises: bool = False) -> pl.DataFrame:
+    lf = (
+        pl.LazyFrame(
+            {
+                "a": ["a", "b", "a", "b", "b", "c"],
+                "b": [1, 2, 3, 4, 5, 6],
+                "c": [6, 5, 4, 3, 2, 1],
+            }
+        )
+        .group_by("a", maintain_order=True)
+        .agg(pl.all().sum())
+    )
+    if raises:
+        lf = lf.select(pl.col("foo_bar"))
+    return await lf.collect_async()
+
+
+async def _aio_collect_all_async(raises: bool = False) -> list[pl.DataFrame]:
+    lf = (
+        pl.LazyFrame(
+            {
+                "a": ["a", "b", "a", "b", "b", "c"],
+                "b": [1, 2, 3, 4, 5, 6],
+                "c": [6, 5, 4, 3, 2, 1],
+            }
+        )
+        .group_by("a", maintain_order=True)
+        .agg(pl.all().sum())
+    )
+    if raises:
+        lf = lf.select(pl.col("foo_bar"))
+
+    lf2 = pl.LazyFrame({"a": [1, 2], "b": [1, 2]}).group_by("a").sum()
+
+    return await pl.collect_all_async([lf, lf2])
+
+
+_aio_collect = pytest.mark.parametrize(
+    ("collect", "raises"),
+    [
+        (_aio_collect_async, None),
+        (_aio_collect_all_async, None),
+        (partial(_aio_collect_async, True), pl.ColumnNotFoundError),
+        (partial(_aio_collect_all_async, True), pl.ColumnNotFoundError),
+    ],
+)
+
+
+def _aio_run(coroutine: Any, raises: Exception | None = None) -> None:
```

Comment:
Have you looked into https://github.com/pytest-dev/pytest-asyncio for running asyncio tests? I find it very helpful to reduce boilerplate code.